save-demo-agent/src/nativeMain/kotlin/com/saveourtool/save/demo/agent/utils/ConfigUtils.kt

```diff
@@ -0,0 +1,45 @@
+/**
+ * Utils to parse save-demo-agent configuration
+ */
+
+package com.saveourtool.save.demo.agent.utils
+
+import com.saveourtool.save.demo.agent.DemoAgentConfig
+
+import okio.FileNotFoundException
+import okio.FileSystem
+import okio.Path.Companion.toPath
+
+import kotlinx.serialization.ExperimentalSerializationApi
+import kotlinx.serialization.properties.Properties
+import kotlinx.serialization.properties.decodeFromStringMap
+
+internal val fs = FileSystem.SYSTEM
+
+/**
+ * Parse config file
+ *
+ * @param configName name of a config file, agent.properties by default
+ * @return [DemoAgentConfig] filled with configuration information
+ */
+@OptIn(ExperimentalSerializationApi::class)
+fun parseConfig(configName: String = "agent.properties"): DemoAgentConfig {
+    val propertiesFile = configName.toPath()
+    require(fs.exists(propertiesFile)) { "Could not find $configName file." }
+    return Properties.decodeFromStringMap(readProperties(propertiesFile.name))
+}
+
+private fun readFile(filePath: String): List<String> = try {
+    val path = filePath.toPath()
+    fs.read(path) { generateSequence { readUtf8Line() }.toList() }
+} catch (e: FileNotFoundException) {
+    emptyList()
+}
+
+private fun readProperties(filePath: String): Map<String, String> = readFile(filePath)
+    .associate { line ->
+        line.split("=").map { it.trim() }.let {
```

Comment:
1. `key: value` is a valid property format, too (`:` can be a separator)
2. You should split only on the 1st separator occurrence, e. g.: `key=va=lue` is a valid entry.
3. Separators (`:` and `=`) can be escaped, and should be un-escaped when reading, e.g.: `path: C\:\\Program Files`

Reply:
May be move this into `cloud-common` and reuse both in  `save-agent` and here?

Reply:
maybe migrate to JSON which supports de-serialization by kotlinx serialization?

Reply:
Seems that kotlinx.serilization is able to serialize `.properties`: https://github.com/Kotlin/kotlinx.serialization/blob/master/formats/README.md

Reply:
> Seems that `kotlinx.serialization` is able to serialize `.properties`

Yep, and this is much better than any wheel we ever reinvent =)

Reply:
> Seems that `kotlinx.serialization` is able to serialize `.properties`: https://github.com/Kotlin/kotlinx.serialization/blob/master/formats/README.md

it does able to serialize, but not de-serialize. It does support de-serialization from `Maps`

Reply:
> > Seems that `kotlinx.serialization` is able to serialize `.properties`: https://github.com/Kotlin/kotlinx.serialization/blob/master/formats/README.md
> 
> it does able to serialize, but not de-serialize. It supports de-serialization from `Maps`

we already have such logic in save-agent

Reply:
> we already have such logic in save-agent

agree and we de-serialize manually to `Map` there and after it we call serialization for properties.
I hope we can migrate JSON or KTOML in save-agent and save-agent-demo

Reply:
> > we already have such logic in save-agent
> 
> agree and we de-serialize manually to `Map` there and after it we call serialization for properties. I hope we can migrate to JSON or KTOML in save-agent and save-agent-demo

toml sounds good 👍

Reply:
@sanyavertolet let's move this logic in common module and migrate to `toml` instead of `.properties`
think migration could be done in separate PR

Reply:
Ok sure